pkg/kn/commands/subscription/update.go

```diff
@@ -62,32 +66,28 @@ func NewSubscriptionUpdateCommand(p *commands.KnParams) *cobra.Command {
 				return err
 			}
 
-			foundSub, err := client.GetSubscription(cmd.Context(), name)
-			if err != nil {
-				return err
-			}
-
-			sb := knmessagingv1.NewSubscriptionBuilderFromExisting(foundSub)
-
-			sub, err := subscriberFlag.ResolveSink(cmd.Context(), dynamicClient, namespace)
-			if err != nil {
-				return err
-			}
-			sb.Subscriber(sub)
-
-			rep, err := replyFlag.ResolveSink(cmd.Context(), dynamicClient, namespace)
-			if err != nil {
-				return err
-			}
-			sb.Reply(rep)
-
-			ds, err := dlsFlag.ResolveSink(cmd.Context(), dynamicClient, namespace)
-			if err != nil {
-				return err
-			}
-			sb.DeadLetterSink(ds)
-
-			err = client.UpdateSubscription(cmd.Context(), sb.Build())
+			err = client.UpdateSubscriptionWithRetry(cmd.Context(), name, func(origSub *messagingv1.Subscription) (*messagingv1.Subscription, error) {
```

Comment:
Maybe I'd opt for the same `updateFunctions := func(){}` definition as per Roland's suggestions and done in other parts of code. But it's really a minor detail.

Reply:
Done